File: back/src/test/java/com/tdevilleduc/urthehero/back/model/StoryTest.java
package com.tdevilleduc.urthehero.back.model;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.util.Collections;

public class StoryTest {

    @Test
    public void test_Constructor() {
        Integer storyId = 67;
        String storyTitle = "test de titre";
        Integer storyAuthorId = 23;
        Integer storyFirstPageId = 98;
        String storyDetailedText = "Tintin long tintin";
        String storyImage = "/chemin/to/image";
        Story story = new Story(storyId, storyTitle, storyAuthorId, storyFirstPageId, storyDetailedText, storyImage);

        Assertions.assertEquals(story.getTitle(), storyTitle);
        Assertions.assertEquals(story.getAuthorId(), storyAuthorId);
        Assertions.assertEquals(story.getFirstPageId(), storyFirstPageId);
        Assertions.assertEquals(story.getDetailedText(), storyDetailedText);
        Assertions.assertEquals(story.getImage(), storyImage);
        Assertions.assertNull(story.getCurrentPageId());
        Assertions.assertTrue(story.getPages().isEmpty());

        Story secondStory = new Story();
        secondStory.setStoryId(storyId);
        secondStory.setTitle(storyTitle);
        secondStory.setAuthorId(storyAuthorId);
        secondStory.setFirstPageId(storyFirstPageId);
        secondStory.setDetailedText(storyDetailedText);
        secondStory.setImage(storyImage);
        secondStory.setPages(Collections.emptyList());

        Assertions.assertEquals(secondStory.toString(), story.toString());
        Assertions.assertEquals(secondStory, story);
    }
}

File: back/src/test/java/com/tdevilleduc/urthehero/back/model/StoryTest.kt
package com.tdevilleduc.urthehero.back.model

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

internal class StoryTest {
    @Test
    fun test_Constructor() {
        val storyId = 42
        val storyTitle = "test de titre"
        val storyAuthorId = 23
        val storyFirstPageId = 98
        val storyDetailedText = "Tintin long tintin"
        val storyImage = "/chemin/to/image"
        val story = Story(storyId, storyAuthorId, storyFirstPageId, storyTitle, storyDetailedText, storyImage)
        Assertions.assertEquals(story.storyId!!, storyId)
        Assertions.assertEquals(story.authorId!!, storyAuthorId)
        Assertions.assertEquals(story.firstPageId!!, storyFirstPageId)
        Assertions.assertEquals(story.title, storyTitle)
        Assertions.assertEquals(story.detailedText, storyDetailedText)
        Assertions.assertEquals(story.image, storyImage)
        Assertions.assertNull(story.currentPageId)
        val secondStory = Story()
        secondStory.storyId = storyId
        secondStory.authorId = storyAuthorId
        secondStory.firstPageId = storyFirstPageId
        secondStory.title = storyTitle
        secondStory.detailedText = storyDetailedText
        secondStory.image = storyImage
        Assertions.assertEquals(secondStory.toString(), story.toString())
        Assertions.assertEquals(secondStory, story)
    }
}